rpm/busybox.spec

```diff
@@ -226,7 +226,7 @@ make distclean
 cp %{SOURCE3} .config
 
 yes "" | make oldconfig
-make %{_smp_mflags}
+%make_build CRYPT_AVAILABLE=n
```

Comment:
Should it link against libcrypt? At least the non static one should I think.

Reply:
Lets not add the dependency here if we do not need the feature, which I assume is the case currently. Thus for my point of view the =n is ok here.

Reply:
Its a depency against libcrypt which is always installed. The buildrequires for libcrypt was just missing.

Reply:
Also we don't want busybox to use its on bundled libcrypt functions.

Reply:
I changed it to work the same way on SDK that it currently works on OBS so basically no change at all. Why don't we want to use busybox's builtin libcrypt functions? Is it mainly a security concern or something else too?

Reply:
There was a change just it wasn't visible since Busybox silently used the builtin crypto, adding pkgconfig libcrypt reinstances the old state before libxcrypt is used (see all the other packages that don't have libcrypt fallbacks for reference)

>Why don't we want to use busybox's builtin libcrypt functions? Is it mainly a security concern or something else too?
Because we don't want to program not using their in tree replacements or why do we have that lib then?
Also there's extra space and so on. I don't see the issue linking against libcrypt as done before.

Reply:
Looks like this is more complicated than I thought. Interestingly leaving out `CRYPT_AVAILABLE=n` from non-static builds is fine. It builds but it also doesn't change the size of the resulting binary at all or add any dependencies to the rpm package. It sounds like the use of internal crypt functions is controlled by `USE_BB_CRYPT` instead (`y` for static, `n` for non-static). However this `CRYPT_AVAILABLE=n` is needed for static builds to succeed on SDK.

Reply:
@Tomin1 Read my comment below, the issue is there's no libxcrypt-static.

Reply:
Yes, I saw your comment and figured out that there is no such thing. I'm just wondering why this doesn't make a difference for non-static builds.

Reply:
Anyway, I skipped that `CRYPT_AVAILABLE=n` for non-static builds now.